pyOCD/tools/gdb_server.py

```diff
@@ -178,9 +179,20 @@ def list_boards(self):
                     'info' : mbed.getInfo(),
                     'board_name' : mbed.getBoardName(),
                     'target' : mbed.getTargetType(),
-                    'vendor_name' : mbed.interface.vendor_name,
-                    'product_name' : mbed.interface.product_name,
+                    'vendor_name' : '',
+                    'product_name' : '',
                     }
+
+                # Reopen the link so we can access the USB vendor and product names from the inteface.
+                # If it's not a USB based link, then we don't attempt this.
+                if isinstance(mbed.link, DAPAccessUSB):
+                    try:
+                        mbed.link.open()
+                        d['vendor_name'] = mbed.link._interface.vendor_name
+                        d['product_name'] = mbed.link._interface.product_name
```

Comment:
No, they're not the same. link.identify() issues a CMSIS-DAP command to get the requested value.
But most CMSIS-DAP builds don't return anything for most of the info keys, except the firmware version. The mbed CMSIS-DAP/DAPLink builds are good examples of this (and they always return "1.0" for the firmware version).

Whether or not that's changed in CMSIS-DAP builds, we'd still need the USB vendor/product string as a backup.